Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+    } else {
+        return wstring(c_defaultUserLanguage);
+    }
+}
+
+/**
+ * Helper method to convert IDWriteLocalizedStrings object to CFString object.
+ *
+ * @parameter localizedString IDWriteLocalizedStrings object to convert.
+ *
+ * @return CFString object.
+ */
+CFStringRef _CFStringFromLocalizedString(IDWriteLocalizedStrings* localizedString) {
+    if (localizedString == NULL) {
+        TraceError(TAG, L"The input parameter is invalid!");
+        return nil;
+    }
+
+    // Get the default locale for this user.
+    wstring localeName = __GetUserDefaultLocaleName();
+
+    uint32_t index = 0;
+    BOOL exists = false;
+
+    // If the default locale is returned, find that locale name, otherwise use "en-us".
+    THROW_IF_FAILED(localizedString->FindLocaleName(localeName.c_str(), &index, &exists));
+    if (!exists) {
+        THROW_IF_FAILED(localizedString->FindLocaleName(c_defaultUserLanguage, &index, &exists));
+    }
+
+    // If the specified locale doesn't exist, select the first on the list.
+    if (!exists) {
+        index = 0;
+    }
+
+    // Get the string length.
+    uint32_t length = 0;
+    THROW_IF_FAILED(localizedString->GetStringLength(index, &length));
+
+    // Get the string.
+    vector<wchar_t> wcharString = std::vector<wchar_t>(length + 1);
+    THROW_IF_FAILED(localizedString->GetString(index, wcharString.data(), length + 1));
+
+    // Strip out unnecessary null terminator
+    return (CFStringRef)CFAutorelease(
+        CFStringCreateWithCharacters(kCFAllocatorSystemDefault, reinterpret_cast<UniChar*>(wcharString.data()), wcharString.size() - 1));
+}
+
+/**
+ * Helper method to retrieve font family names installed in the system.
+ *
+ * @return Unmutable array of font family name strings that are installed in the system.
+ */
+CFArrayRef _DWriteGetFontFamilyNames() {
+    CFMutableArrayRef fontFamilyNames = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
+    CFAutorelease(fontFamilyNames);
+
+    // Get the direct write factory instance
+    ComPtr<IDWriteFactory> dwriteFactory;
+    THROW_IF_FAILED(_DWriteCreateFactoryInstance(&dwriteFactory));
+
+    // Get the system font collection.
+    ComPtr<IDWriteFontCollection> fontCollection;
+    THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
+
+    // Get the number of font families in the collection.
+    uint32_t count = 0;
+    count = fontCollection->GetFontFamilyCount();
+
+    for (uint32_t i = 0; i < count; ++i) {
+        // Get the font family.
+        ComPtr<IDWriteFontFamily> fontFamily;
+        THROW_IF_FAILED(fontCollection->GetFontFamily(i, &fontFamily));
+
+        // Get a list of localized strings for the family name.
+        ComPtr<IDWriteLocalizedStrings> familyNames;
+        THROW_IF_FAILED(fontFamily->GetFamilyNames(&familyNames));
+
+        CFStringRef name = _CFStringFromLocalizedString(familyNames.Get());
+        if (CFStringGetLength(name) == 0) {
```

Comment:
sure

---
In reply to: [84339235](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r84339235) [](ancestors = 84339235)